src/util/type.h

```diff
@@ -203,4 +203,8 @@ pre-defined types:
 bool is_number(const typet &type);
 // rational, real, integer, complex, unsignedbv, signedbv, floatbv
 
+// If a given type is a constant variable (e.g const int) or a constant
+// array (e.g. const int [])
```

Comment:
Please word carefully: what does it mean for a type to be a "constant variable"? And: In C, `const int []` does not actually declare a constant array, but instead declares an array of constant `int` elements. (This, however, implies that the entire object is constant as the size is constant.)